vm/src/instruction.rs

```diff
@@ -36,3 +36,14 @@ pub enum Instruction {
     Ripemd160,
     Keccak256,
 }
+
+pub fn is_valid_unlock_script(instrs: &[Instruction]) -> bool {
+    for instr in instrs {
+        match instr {
+            Instruction::Push(_) => {}
+            Instruction::PushB(_) => {}
+            _ => return false,
+        }
+    }
+    true
```

Comment:
I think we can use `any` or `all` in `Iterator`.
e.g. `instrs.iter().all(|inst| ...)`

Reply:
I tried it, It's not possible. 
would you advise me?

Reply:
Thank you, but I'm not sure what's better than the old one.

Reply:
I initially thought it would become simpler, but it seems not for this case.
I'd prefer functional style, but I guess it's up to you.
I'll accept either `iter` or `for` syntax for here.